.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,51 @@
+name: Build and Publish
+
+on:
+  push:
+    branches:
+    - main
+
+env:
+  MAJOR: 4
+  MINOR: 0
+  MINORMINOR: 1 
+  RUN: ${{ github.run_number }}
+
+jobs:
+  build:
+    runs-on: windows-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v2
+
+      - name: Restore dependencies
+        run: |
+          nuget sources add -username Open-Systems-Pharmacology -password ${{ secrets.GITHUB_TOKEN }} -name OSP-GitHub-Packages -source "https://nuget.pkg.github.com/Open-Systems-Pharmacology/index.json"
+          nuget restore
+
+      - name: define env variables
+        run: |
+          echo "APP_VERSION=${{ env.MAJOR }}.${{ env.MINOR }}.${{ env.MINORMINOR }}.${{ env.RUN }}" | Out-File -FilePath $env:GITHUB_ENV -Append
```

Comment:
Somehow I miss the part with getting the version of the last published package version and setting the new version accordingly.

Reply:
Instead of doing that, I just incremented the Minor-minor version number and use the run number directly.

Reply:
ah ok, you can use the run number because you separated the workflows on pr and on push